### PR TITLE
[BEAM-14236] Parquet IO support for list to conform with Apache Parqu…

### PR DESCRIPTION
**Pull request by Shiv22Wabale**
…et specification for Python SDK

Added flag driven apache parquet conform write for type list.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Shiv22Wabale** [comment]
run python postcommit

**Comment by Shiv22Wabale** [comment]
run python postcommit

**Comment by Shiv22Wabale** [comment]
retest this please

**Comment by Shiv22Wabale** [comment]
retest this please

**Comment by Shiv22Wabale** [comment]
retest this please

**Comment by Shiv22Wabale** [comment]
R: @ihji

**Comment by Shiv22Wabale** [comment]
R: @ihji

**Comment by ihji** [comment]
@Shiv22Wabale You can use `./gradlew lint` to check lint errors before uploading the PR.

**Comment by Shiv22Wabale** [comment]
R: @ihji

File: sdks/python/apache_beam/io/parquetio_test.py [hunk 1]
```diff
@@ -109,6 +111,37 @@ def setUp(self):
...
+                'favorite_color': 'blue'
+            },
+            {
+                'name': 'Henry',
+                'favorite_number': 3,
+                'favorite_color': 'green'
+            },
+        ]
+    },
+                           {
```
Comment (ihji) [inline sdks/python/apache_beam/io/parquetio_test.py:128]: Is this a correct indentation? Shouldn't this be the same level with the previous `items`?

File: sdks/python/apache_beam/io/parquetio_test.py [hunk 2]
```diff
@@ -314,6 +348,28 @@ def test_sink_transform(self):
...
+        _ = p \
+        | Create(self.RECORDS_NESTED) \
+        | WriteToParquet(
+            path, self.SCHEMA_NESTED, num_shards=1,
+            shard_name_template='', use_compliant_nested_type=True)
+      with TestPipeline() as p:
+        # json used for stable sortability
+        readback = \
+            p \
+            | ReadFromParquet(path) \
```
Comment (ihji) [inline sdks/python/apache_beam/io/parquetio_test.py:368]: Just for clarification, do we need additional `use_compliant_nested` option only for the writer not for the reader? So do you plan to use the consumer other than `ReadFromParquet`?

**Review by ihji — COMMENTED** [review]
Thanks, looks great.

File: sdks/python/apache_beam/io/parquetio_test.py (hunk 1 above)
Comment (Shiv22Wabale) [reply sdks/python/apache_beam/io/parquetio_test.py:128]: Yes, it is correct from PythonFormatter.

If I keep the it with same level with the previous item, the PythonFormatter PreCommit fails.

File: sdks/python/apache_beam/io/parquetio_test.py (hunk 2 above)
Comment (Shiv22Wabale) [reply sdks/python/apache_beam/io/parquetio_test.py:368]: We just need it for writer. 
The ReadFromParquet would just use the schema in the parquet file and doesn't require any external indicator/flag to read the parquet with compliant nested fields.

And we'll be using different consumer other than ReadFromParquet but I don't think we need any changes from reader side.

**Comment by Shiv22Wabale** [comment]
R: @ihji

**Review by ihji — APPROVED** [review]

**Pull request merged**